### DISCUSSION TITLE
How to write a boolean expression for prisma query filters?

### DISCUSSION THREAD

**Discussion by RitikDua** [prisma/prisma#3205] [Q&A] [answered]
Suppose we have a data model of name "sample" containing "value" as int. Thus, Prisma model is as follow :
```prisma
model sample{
value:Int
}
```
Graphql schema is as follow:
```graphql
type Query{
   find(minvalue:Int,maxvalue:Int):sample!
}
type sample{
   value:Int
}
```
Now i would like to perform a prisma query such as:
**(value >= minprice && value <= maxprice)**

I tried methods like this below which couldn't work:
```ts
async find(parent,args,ctx,info){
    const filter=args.filter
          ?{AND:[
                  {value:{
                    _lte:args.maxvalue,
                  }},
                {value:{
                      _gte:args.minvalue,
                  }},
                ],
            }:{}
    return await ctx.prisma.sample.findMany({
              filter
    });
}
```

**Comment by ryands17** [comment, marked as answer]
Hey @RitikDua 👋 
You could do it like this:

```ts
prisma.sample.findMany({
    where: {
      AND: [{ value: { gte: 1 } }, { value: { lte: 5 } }],
    },
})
```

**Reply by RitikDua** [reply]
Hey @ryands17, I had tried this method earlier and it was also not working.
P.S. Thanks for your reply.

**Reply by ryands17** [reply]
Could you share the error?

**Reply by RitikDua** [reply]
Actually, it is not an error.
I have a database that stores the different values of "value"  and on passing the query it is showing them all.

**Reply by ryands17** [reply]
I inserted values and queried for the data in this way:

```ts
for (let value of [10, 20, 30, 40, 50, 60]) {
    await prisma.sample.create({
      data: {
        value,
      },
    })
  }

  const res = await prisma.sample.findMany({
    where: {
      AND: [{ value: { gte: 20 } }, { value: { lte: 50 } }],
    },
  })
```

And this is the result

```js
[
  { id: 2, value: 20 },
  { id: 3, value: 30 },
  { id: 4, value: 40 },
  { id: 5, value: 50 }
]
```

So could you check the data and query that you are making and let me know if it is the same as above. Also I am on version `2.4.0` of `@prisma/client` and `@prisma/cli`.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
